src/test/cpp/gtester/CheckPointTest.cpp

```diff
@@ -0,0 +1,37 @@
+#include <iostream>
+#include <vector>
+#include <checkpoint/CheckPoint.h>
+#include <gtest/gtest.h>
+#include <stdio.h>
+#include <boost/filesystem.hpp>
+
+
+namespace Tests {
+
+TEST(CheckPoint, CheckPoint)
```

Comment:
AFAICT, these tests don't actually ensure that checkpointing really works. They just create/open/close files. Can you add some tests that actually write a checkpoint to a file, or even an in-memory stream?